File: src/components/Order.java
/ Example in BidOrder.tsx

interface Order {
  id: string;
  price: number;
  quantity: number;
  side: 'bid' | 'ask';
}

interface BidOrderProps {
  addOrder: (order: Order) => void;
}

const BidOrder: React.FC<BidOrderProps> = ({ addOrder }) => {
  // component logic here...
}

File: src/types/order.ts
// types/order.ts

import {createAsyncThunk} from "@reduxjs/toolkit";

export interface Order {
    id: string;
    price: number;
    quantity: number;
    side: 'bid' | 'ask';
}

// types/order.ts

export interface OrderInput {
    price: number;
    quantity: number;
    side: 'bid' | 'ask';
}

export const addOrder = createAsyncThunk(
    'orders/addOrder',
    async (order: OrderInput) => {
        // ...
    }
);
